src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -2296,14 +2297,10 @@ public override BoundNode VisitConditionalAccess(BoundConditionalAccess node)
             }
             else
             {
-                // In the right-hand-side, the left-hand-side is known to be non-null.
-                int slot = MakeSlot(SkipReferenceConversions(receiver));
-                _lastConditionalAccessSlot = slot;
-                if (slot > 0)
-                {
-                    if (slot >= this.State.Capacity) Normalize(ref this.State);
-                    this.State[slot] = NullableFlowState.NotNull;
-                }
+                // In the when-null branch, the receiver is known to be maybe-null.
+                // In the other branch, the receiver is known to be non-null.
+                _lastConditionalAccessSlot = LearnFromNullTest(receiver, ref receiverState);
+                LearnFromNonNullTest(receiver, ref this.State);
```

Comment:
Consider using a discard here `_ = ` to make it explicit the value is ignored.

Reply:
That method doesn't return a value. I only updated one of the two methods to return the slot, based on need so far.